host-macros/src/service.rs

```diff
@@ -147,17 +198,24 @@ impl ServiceBuilder {
             .on_write
             .as_ref()
             .map(|callback| quote!(builder.set_write_callback(#callback);));
+        let default_value = match characteristic.args.default_value {
+            Some(val) => quote!(#val),      // if set by user
+            None => quote!(#ty::default()), // or default otherwise
```

Comment:
I wonder if we should constrain `GattValue` and `FixedGattValue` to be `Default`? Otherwise one might have some interesting errors if they used a type which doesn't implement `Default`

Reply:
Looks like we can't do this, because of impl GattValue for [u8; N]

So might need to change back what we do in the None case.

Reply:
Yeah I think we'll have to revert it, otherwise [u8; N] is unusable.

Reply:
solved! `<#ty>::default()`